contracts/HydraDelegation/DelegationPoolLib.sol

```diff
@@ -159,19 +209,152 @@ library DelegationPoolLib {
      * @notice claims the rewards for an address in a pool
      * @param pool the DelegationPool to claim the rewards from
      * @param account the address for which to claim the rewards
-     * @param rps the reward per share
-     * @param balance the balance of the account
-     * @param correction the correction of the account
+     * @param epochNumber Epoch where the last claimable reward is distributed
+     * We need it because not all rewards are matured at the moment of claiming
+     * @param balanceChangeIndex Whether to redelegate the claimed rewards
      * @return reward the amount of rewards claimed
      */
     function claimRewards(
         DelegationPool storage pool,
         address account,
-        uint256 rps,
-        uint256 balance,
-        int256 correction
+        uint256 epochNumber,
+        uint256 balanceChangeIndex
     ) internal returns (uint256 reward) {
-        reward = claimableRewards(pool, account, rps, balance, correction);
+        reward = claimableRewards(pool, account, epochNumber, balanceChangeIndex);
         pool.claimedRewards[account] += reward;
     }
+
+    /**
+     * @notice Deletes  the historical data of a delegator
+     * @param pool the DelegationPool to clean the historical data from
+     * @param account the address of the  delegator to clean the historical data from
+     */
+    function cleanDelegatorHistoricalData(DelegationPool storage pool, address account) internal {
```

Comment:
Would it be good idea to clean the data when swapping, for the new validator

Reply:
GC!